index.js

```diff
@@ -189,3 +210,37 @@ function addStartupSupportabilities(agent) {
     }
   }
 }
+
+/**
+ * Used to determine how the agent is getting loaded:
+ *  1. -r newrelic
+ *  2. --loader newrelic/esm-loader.mjs
+ *  3. require('newrelic')
+ *
+ *  Then a supportability metric is loaded to decide.
+ *  Note: We already take care of scenario #2 in newrelic/esm-loader.mjs
```

Comment:
Just a thought, but would it be worth moving that function from the loader into here and looking for the cli args, that way we have one place to go as opposed to two?

Reply:
I'm not a fan just because it's more involved to detect if our agent is getting loaded this way.